rmf_site_editor/src/site/model.rs

```diff
@@ -368,6 +432,23 @@ impl<'w, 's> ModelLoader<'w, 's> {
                 .instruct(SpawnModelLabel(entity).preempt()),
         )
     }
+
+    /// Update the asset source of all model instances affiliated with the provided
+    /// model description
+    pub fn update_description_asset_source(&mut self, entity: Entity, source: AssetSource) {
+        let mut instance_entities = HashSet::new();
+        for (e, affiliation) in self.model_instances.iter() {
+            if let Some(description_entity) = affiliation.0 {
+                if entity == description_entity {
+                    instance_entities.insert(e);
+                }
+            }
+        }
+        for e in instance_entities.iter() {
+            self.update_asset_source_impulse(*e, source.clone())
+                .detach();
+        }
+    }
```

Comment:
Can this be simplified to remove the intermediate `HashSet` and just fire the workflow inside the first for loop? From a glance it doesn't seem there should be cases where duplicated entities are added (which would justify collecting into a hash set)

Reply:
I was unable to borrow a mutable `self` for `self.update_asset_source_impulse` while iterating over `self.model_instances`

Reply:
Ah yes I see it now :facepalm: nevermind, thanks for clarifying!